### ISSUE TITLE
[Compilers] figure out how to implement `linker` flag in solidity compiler in deployment

### ISSUE THREAD

**Issue by VoR0220** [eris-ltd/eris#1193]
The linker helps us to automatically link the stubs in binaries directly to addresses of libraries that are already deployed without revealing source code. Take time to play with this and see how we can implement it in the new compilers setup and (if easy to do) apply it to old compilers in patch version.

**Comment by j-h-scheufen** [comment]
Requirements:
- Ability to deploy binary contracts that have not been linked and invoke linker during deployment, yet. I.e. they still show the library placeholders within the binary content: `09150618981565b73__BaseErrors____________________________635f1c4c0e604051817c`

Functionality:
- It is currently already possible to deploy a binary contract by specifying the `.bin` file in the EPM deploy job
- The deploy job should internally invoke the linker as an additional step before the deployment if
  a) a binary file is being deployed
  b) the `libraries` attribute has been set

**Comment by VoR0220** [comment]
So I struggled to figure out how to get this working and I found out it had a bug in it (relatively easy fix but still kind of a pain that we aren't standardizing a command line testing suite...will have to bring that up with Chris). 

Something to keep in mind:

the linker can only be invoked for .bin files or an output of the binary. For source code, it will absolutely fail every single time. 

Currently I have found only one real correct way of doing this (and this is with my fix) insofar as command line usage. However there are two different approaches we can use.

A) We pass it in as a process, for example, using the libraries test fixture:
```
solc consuming-contract.sol --combined-json bin | solc --link --libraries single-lib.sol:Search:0x692a70d2e424a56d2c6c27aa97d1a86395877b3a
```
B) we generate the .bin files before hand using ```solc -o binaries --bin consuming-contract.sol``` and from there run ```cat C.bin | solc --link --libraries single-lib.sol:Search:0x692a70d2e424a56d2c6c27aa97d1a86395877b3a```

In conclusion I think we would be better suited to only invoking the linker when `.bin` files are passed in, and they too would require a `libraries` reference.

**Comment by j-h-scheufen** [comment]
@VoR0220 Sorry, just noticing your comments above. Please see the requirements description. We ONLY need the linking for binary files right now. For .sol files it is already working as expected and no changes required!

**Comment by VoR0220** [comment]
Basically closed, just waiting on a merge.
